src/routes/edit-invoice.tsx

```diff
@@ -6,49 +6,61 @@ import { DatePicker } from '@mui/x-date-pickers';
 import { BillingForm } from '../components/BillingForm.tsx';
 import { OrderLinesForm } from '../components/OrderLinesForm.tsx';
 import { zodResolver } from '@hookform/resolvers/zod';
-import InvoiceSchema from '../types/Invoice.ts';
-import InvoicesService from '../services/invoices/invoicesService.ts';
-import { useEffect, useState } from 'react';
-import { useNavigate, useParams } from 'react-router-dom';
-import { useQuery } from 'react-query';
+import { useState } from 'react';
+import { Link, useMatch, useParams } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
-import dayjs from 'dayjs';
+import { useGetInvoice, useUpdateInvoice } from '../hooks/invoices.hooks.ts';
+import { z } from 'zod';
+import InvoiceSchema from '../types/Invoice.ts';
 
-export default function ViewInvoice() {
-  const { t } = useTranslation();
-  const { id } = useParams<{ id: string }>();
-  const { handleSubmit, register, control, setValue } = useForm({
-    resolver: zodResolver(InvoiceSchema),
-  });
-  const navigate = useNavigate();
+interface ViewInvoiceProps {
+  defaultValues?: any;
+  isEditMode?: boolean;
+}
 
-  const [status, setStatus] = useState('idle');
-  const [isEditMode, setIsEditMode] = useState(false);
+const useIsEditMode = () => {
+  const match = useMatch('/invoice/:id/edit');
```

Comment:
This is fine, but I would rather pass a prop in router. That way you have only one source of truth, while here you need to sync between router and component (what if someone changes the path in router and doesn't know about this hook?)